services/src/main/java/org/keycloak/broker/oidc/mappers/ClaimToUserSessionNoteMapper.java

```diff
@@ -0,0 +1,123 @@
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+public class ClaimToUserSessionNoteMapper extends AbstractClaimMapper {
+
+    private static final String CLAIM_PROPERTY_NAME = "claims";
+    private static final String[] COMPATIBLE_PROVIDERS = {KeycloakOIDCIdentityProviderFactory.PROVIDER_ID,
+            OIDCIdentityProviderFactory.PROVIDER_ID};
+
+    private static final List<ProviderConfigProperty> CONFIG_PROPERTIES = new ArrayList<>();
+
+    private static final Set<IdentityProviderSyncMode> IDENTITY_PROVIDER_SYNC_MODES =
+            new HashSet<>(Arrays.asList(IdentityProviderSyncMode.values()));
+
+    static {
+        ProviderConfigProperty claimsProperty = new ProviderConfigProperty();
+        claimsProperty.setName(CLAIM_PROPERTY_NAME);
+        claimsProperty.setLabel("Claims");
+        claimsProperty.setHelpText(
+                "Name of the claims to search for in the token. You can reference nested claims using a '.', i.e. 'address.locality'. To use dot (.) literally, escape it with backslash (\\.)");
+        claimsProperty.setType(ProviderConfigProperty.MULTIVALUED_STRING_TYPE);
+        CONFIG_PROPERTIES.add(claimsProperty);
+    }
+
+    public static final String PROVIDER_ID = "oidc-user-session-note-idp-mapper";
+
+    @Override
+    public String[] getCompatibleProviders() {
+        return COMPATIBLE_PROVIDERS;
+    }
+
+    @Override
+    public String getDisplayCategory() {
+        return "User Session";
+    }
+
+    @Override
+    public String getDisplayType() {
+        return "User Session Note Mapper";
+    }
+
+    @Override
+    public String getHelpText() {
+        return "Add every matching claim to the user session note.";
+    }
+
+    @Override
+    public List<ProviderConfigProperty> getConfigProperties() {
+        return CONFIG_PROPERTIES;
+    }
+
+    @Override
+    public String getId() {
+        return PROVIDER_ID;
+    }
+
+    @Override
+    public boolean supportsSyncMode(IdentityProviderSyncMode syncMode) {
+        return IDENTITY_PROVIDER_SYNC_MODES.contains(syncMode);
+    }
+
+    @Override
+    public void importNewUser(KeycloakSession session, RealmModel realm, UserModel user,
+            IdentityProviderMapperModel mapperModel, BrokeredIdentityContext context) {
+        addClaimToSessionNote(mapperModel, context);
+    }
+
+    @Override
+    public void updateBrokeredUser(KeycloakSession session, RealmModel realm, UserModel user,
+            IdentityProviderMapperModel mapperModel, BrokeredIdentityContext context) {
+        addClaimToSessionNote(mapperModel, context);
+    }
+
+    private void addClaimToSessionNote(IdentityProviderMapperModel mapperModel, BrokeredIdentityContext context) {
+        List<String> claims = getConfigList(mapperModel.getConfig(), CLAIM_PROPERTY_NAME);
+
+        for (String claim : claims) {
+            Object valueObj = getClaimValue(context, claim);
+
+            if (valueObj == null) {
+                break;
```

Comment:
Why `break`? All other claims would be discarded and not mapped if one has a null value.

Reply:
Good catch, should be `continue`.

Reply:
I refactored the code so it uses only one `continue` (sonar finding).